pkg/api/types.go

```diff
@@ -2070,10 +2070,41 @@ const (
 	StrategicMergePatchType PatchType = "application/strategic-merge-patch+json"
 )
 
+// Type and constants for component type.
+type ComponentType string
+
+// These are the valid conditions for the component type.
+const (
+	ComponentAPIServer         ComponentType = "apiserver"
+	ComponentControllerManager ComponentType = "controller-manager"
+	ComponentScheduler         ComponentType = "scheduler" // TODO: move mesos-specific component type & validation to contrib
```

Comment:
These constants should be made to adhere to the CamelCase convention for API constants, which will be possible once they are no longer used as name prefixes.